app/src/main/java/ch/epfl/favo/view/tabs/MapsPage.java

```diff
@@ -124,32 +137,50 @@ public View onCreateView(
     FloatingActionButton button = view.findViewById(R.id.offline_map_button);
     button.setOnClickListener(this::onOfflineMapClick);
 
+    RadioButton toggle = view.findViewById(R.id.list_switch);
+    toggle.setOnClickListener(this::onToggleClick);
+    activity = (MainActivity) Objects.requireNonNull(getActivity());
+    setupView();
+    SupportMapFragment mapFragment = (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
+    if (mapFragment != null)
+      mapFragment.getMapAsync(this);
+    mLocation = DependencyFactory.getCurrentGpsTracker(getContext()).getLocation();
+    updateNearbyList();
     return view;
   }
 
-  @Override
-  public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
-    super.onViewCreated(view, savedInstanceState);
-    setupView();
-    SupportMapFragment mapFragment =
-        (SupportMapFragment) getChildFragmentManager().findFragmentById(R.id.map);
-    mGpsTracker = new GpsTracker(getContext());
-    if (mapFragment != null) {
-      mapFragment.getMapAsync(this);
+  private void updateNearbyList(){
+    String setting = activity.getPreferences(Context.MODE_PRIVATE).getString("radius", "1km");
+    double radius = 1;
+    switch (setting){
+      case "1 Km":
```

Comment:
Can maybe use regexp instead of switch

Reply:
I am rewriting it now

Reply:
done with split()
String setting = activity.getPreferences(Context.MODE_PRIVATE).getString("radius", "1 Km");
double radius = Double.parseDouble(setting.split(" ")[0]);

Reply:
Looks good